move/sources/food.move

```diff
@@ -0,0 +1,161 @@
+    /// The food collection description
+    const FOOD_COLLECTION_DESCRIPTION: vector<u8> = b"Food Collection Description";
+    /// The food collection URI
+    const FOOD_COLLECTION_URI: vector<u8> = b"https://food.collection.uri";
+    const FOOD_DESCRIPTION: vector<u8> = b"Food Description";
+    const FOOD_URI: vector<u8> = b"https://otjbxblyfunmfblzdegw.supabase.co/storage/v1/object/public/aptogotchi/food.png";
+    const FOOD_NAME: vector<u8> = b"Food";
+    const PROJECT_URI: vector<u8> = b"https://www.aptoslabs.com";
+
+
+    struct FoodToken has key {
+        /// Used to mint fungible assets.
+        fungible_asset_mint_ref: fungible_asset::MintRef,
+        /// Used to burn fungible assets.
+        fungible_asset_burn_ref: fungible_asset::BurnRef,
+    }
+
+    fun init_module(account: &signer) {
+        create_food_collection(account);
+
+        create_food_token(
+            account,
+            string::utf8(FOOD_DESCRIPTION),
+            string::utf8(FOOD_NAME),
+            string::utf8(FOOD_URI),
+            string::utf8(FOOD_NAME),
+            string::utf8(FOOD_NAME),
+            string::utf8(FOOD_URI),
+            string::utf8(PROJECT_URI),
+        );
+    }
+
+    /// Creates the food collection.
+    fun create_food_collection(creator: &signer) {
+        // Constructs the strings from the bytes.
+        let description = string::utf8(FOOD_COLLECTION_DESCRIPTION);
+        let name = string::utf8(FOOD_COLLECTION_NAME);
+        let uri = string::utf8(FOOD_COLLECTION_URI);
+
+        collection::create_unlimited_collection(
+            creator,
+            description,
+            name,
+            option::none(),
+            uri,
+        );
+    }
+
+    /// Creates the food token as fungible token.
+    fun create_food_token(
+        creator: &signer,
+        description: String,
+        name: String,
+        uri: String,
+        fungible_asset_name: String,
+        fungible_asset_symbol: String,
+        icon_uri: String,
+        project_uri: String,
+    ) {
+        let collection = string::utf8(FOOD_COLLECTION_NAME);
+        let constructor_ref = token::create_named_token(
+            creator,
+            collection,
+            description,
+            name,
+            option::none(),
+            uri,
+        );
+
+        let token_signer = object::generate_signer(&constructor_ref);
+
+        // Creates the fungible asset.
+        primary_fungible_store::create_primary_store_enabled_fungible_asset(
+            &constructor_ref,
+            option::none(),
+            fungible_asset_name,
+            fungible_asset_symbol,
+            0,
+            icon_uri,
+            project_uri,
+        );
+        let fungible_asset_mint_ref = fungible_asset::generate_mint_ref(&constructor_ref);
+        let fungible_asset_burn_ref = fungible_asset::generate_burn_ref(&constructor_ref);
+
+        // Publishes the FoodToken resource with the refs.
+        let food_token = FoodToken {
```

Comment:
nit: can be combined with the move_to call:

move_to(&token_signer, FoodToken {
  ...
});